mesheryctl/internal/cli/root/app/view.go

```diff
@@ -0,0 +1,133 @@
+package app
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	viewAllFlag   bool
+	outFormatFlag string
+)
+
+var viewCmd = &cobra.Command{
+	Use:   "view <application name>",
+	Short: "Display application(s)",
+	Long:  `Displays the contents of a specific application based on name or id`,
+	Args:  cobra.MaximumNArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return errors.Wrap(err, "error processing config")
+		}
+		application := ""
+		isID := false
+		applicationID := ""
+		// if application name/id available
+		if len(args) > 0 {
+			if viewAllFlag {
+				return errors.New("-a cannot be used when [application-name|application-id] is specified")
+			}
+			applicationID = args[0]
+			// check if the application argument is a valid uuid v4 string
+			isID, err = regexp.MatchString("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$", applicationID)
+			if err != nil {
+				return err
+			}
+		}
+		url := mctlCfg.GetBaseMesheryURL()
+		if len(application) == 0 {
```

Comment:
length of application will be always zero, because of that we get this.
![image](https://user-images.githubusercontent.com/50979887/126112673-619b2e7d-f034-4231-9c57-19dd6f4d2fcb.png)

Reply:
Good point